### PR TITLE
Enables resource pack loading in Dragonfly

### PR DESCRIPTION
**Pull request by SALLYPEMDAS**
https://github.com/df-mc/dragonfly/issues/255

### CODE REVIEW

File: server/listener.go [hunk 1]
```diff
@@ -1,9 +1,11 @@
 package server
 
 import (
+	"io"
+
 	"github.com/df-mc/dragonfly/server/session"
 	"github.com/sandertv/gophertunnel/minecraft"
-	"io"
+	"github.com/sandertv/gophertunnel/minecraft/resource"
```
Comment (SALLYPEMDAS) [inline server/listener.go:8]: reverting these changes

File: server/server.go [hunk 2]
```diff
@@ -94,6 +98,28 @@ func New(c *Config, log internal.Logger) *Server {
...
+	if _, err := os.Stat(c.Resources.Folder); os.IsNotExist(err) {
+		_ = os.Mkdir(c.Resources.Folder, 0777)
+	}
+	resources, err := os.ReadDir(c.Resources.Folder)
+	if err != nil {
+		panic(err)
+	}
+	for _, entry := range resources {
+		fmt.Println(" ", entry.Name(), entry.IsDir())
+		fmt.Println(entry.Name())
```
Comment (DaPigGuy) [inline server/server.go:110]:
```suggestion
```

Comment (SALLYPEMDAS) [reply server/server.go:110]: will fix

File: server/server.go [hunk 3]
```diff
@@ -94,6 +98,26 @@ func New(c *Config, log internal.Logger) *Server {
...
 
+	if _, err := os.Stat(c.Resources.Folder); os.IsNotExist(err) {
+		_ = os.Mkdir(c.Resources.Folder, 0777)
+	}
+	resources, err := os.ReadDir(c.Resources.Folder)
+	if err != nil {
+		panic(err)
+	}
+	for _, entry := range resources {
+		if entry.IsDir() {
```
Comment (TwistedAsylumMC) [inline server/server.go:109]: Why ignore directories here? Gophertunnel's resource pack implementation can parse directories too.

Comment (SALLYPEMDAS) [reply server/server.go:109]: Thanks for the review! Will fix!

**Review by Sandertv — CHANGES_REQUESTED** [review]
Thanks for the PR!

Currently this doesn't have an entry added in the config.toml. Could you add it there with the appropriate documentation? Your PR looks really good otherwise, you could also consider moving the resource pack loading in server.New() to a separate function.

**Comment by SALLYPEMDAS** [comment]
Thanks for the reviews!!!

PR updated:

Let me know if there are any more changes :)

File: server/server.go [hunk 4]
```diff
@@ -547,3 +553,23 @@ func (server *Server) itemEntries() (entries []protocol.ItemEntry) {
 	}
 	return
 }
+
+// loadResources loads resource packs from path of specifed directory
```
Comment (Sandertv) [inline server/server.go:557]: `.` at the end here please!

File: server/config.go [hunk 5]
```diff
@@ -49,6 +49,11 @@ type Config struct {
 		// player provider if it is enabled.
 		Folder string
 	}
+
+	Resources struct {
+		// Folder controls the location where resource packs will be loaded from
```
Comment (Sandertv) [inline server/config.go:54]: `.` at the end here please!

File: config.toml [hunk 6]
```diff
@@ -45,3 +45,7 @@
   # Folder controls where the player data will be stored by the default LevelDB
   # player provider if it is enabled.
   Folder = "players"
+
+[Resources]
+  # Folder configures the directory used by the server to load resource packs
```
Comment (Sandertv) [inline config.toml:50]: `.` at the end here please!

**Review by Sandertv — COMMENTED** [review]
Last couple of documentation nitpicks, ready to go after this!

File: config.toml (hunk 6 above)
Comment (SALLYPEMDAS) [reply config.toml:50]: certainly!

File: server/config.go (hunk 5 above)
Comment (SALLYPEMDAS) [reply server/config.go:54]: Yes!

File: server/server.go (hunk 4 above)
Comment (SALLYPEMDAS) [reply server/server.go:557]: 👍

**Review by Sandertv — APPROVED** [review]
Looks good, thanks again for your PR!

**Comment by SALLYPEMDAS** [comment]
<3

**Pull request merged**